pkg/controller/che/che_controller.go

```diff
@@ -1191,6 +1155,179 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
+			cheManageNameSpacesRole := deploy.NewClusterRole(instance, cheManageNamespacesName, []rbac.PolicyRule{
+				{
+					APIGroups: []string{"authorization.openshift.io", "rbac.authorization.k8s.io"},
+					Resources: []string{"roles"},
+					Verbs: []string{"get", "create"},
+				},
+				{
+					APIGroups: []string{"authorization.openshift.io", "rbac.authorization.k8s.io"},
+					Resources: []string{"rolebindings"},
+					Verbs: []string{"get", "update", "create"},
+				},
+				{
+					APIGroups: []string{"project.openshift.io"},
+					Resources: []string{"projects"},
+					Verbs: []string{"get"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"serviceaccounts"},
+					Verbs: []string{"get", "create", "watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods/exec"},
+					Verbs: []string{"create"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"persistentvolumeclaims", "configmaps"},
+					Verbs: []string{"list"},
+				},
+				{
+					APIGroups: []string{"apps"},
+					Resources: []string{"secrets"},
+					Verbs: []string{"list"},
+				},
+
+				{
+					APIGroups: []string{""},
+					Resources: []string{"secrets"},
+					Verbs: []string{"list", "create", "delete"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"persistentvolumeclaims"},
+					Verbs: []string{"get", "create", "watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods"},
+					Verbs: []string{"get", "create", "list", "watch", "delete"},
+				},
+				{
+					APIGroups: []string{"apps"},
+					Resources: []string{"deployments"},
+					Verbs: []string{"get", "create", "list", "watch", "patch", "delete"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"services"},
+					Verbs: []string{"create", "list", "delete"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"configmaps"},
+					Verbs: []string{"create", "delete"},
+				},
+				{
+					APIGroups: []string{"route.openshift.io"},
+					Resources: []string{"routes"},
+					Verbs: []string{"list", "create", "delete"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"events"},
+					Verbs: []string{"watch"},
+				},
+				{
+					APIGroups: []string{"apps"},
+					Resources: []string{"replicasets"},
+					Verbs: []string{"list", "get", "patch", "delete"},
+				},
+				{
+					APIGroups: []string{"extensions"},
+					Resources: []string{"ingresses"},
+					Verbs: []string{"list", "create", "watch", "get", "delete"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"namespaces"},
+					Verbs: []string{"get"},
+				},
+			})
+			if err := r.CreateNewClusterRole(instance, cheManageNameSpacesRole); err != nil {
+				return err
+			}
+
+			// this binding is needed to manage che workspaces out of che namespace
+			// `${namespace}-clusterrole-manage-namespaces` ClusterRole should be created during che-operator deploy
```

Comment:
What do you mean ? That it is just created 5 lines above ?

Reply:
We delegate some part permissions to the che server